crates/shared/src/app_data.rs

```diff
@@ -18,10 +21,17 @@ pub struct ValidatedAppData {
 }
 
 #[derive(Debug, Default, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
 pub struct ProtocolAppData {
     #[serde(default)]
     pub hooks: Hooks,
     pub signer: Option<H160>,
+    pub replaced_order: Option<ReplacedOrder>,
```

Comment:
Should we maybe throw an error if this value is populated while placing an order with the regular endpoint?
Actually if this we are now always parsing this field from the metadata we could even consider ditching the `PATCH /orders` endpoint altogether and only implement the replacement mechanism on `POST /orders` with that special appdata. That would get rid of the ambiguity.